### PR TITLE
mcs: add more tso tests

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #5836.

### What is changed and how does it work?
Add more tests for tso.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: tests/mcs/testutil.go [hunk 1]
```diff
@@ -37,6 +37,13 @@ func SetupTSOClient(ctx context.Context, re *require.Assertions, endpoints []str
 	return cli
 }
 
+// SetupTSOClient creates a TSO client for test.
```
Comment (binshi-bing) [inline tests/mcs/testutil.go:40]:
```suggestion
// SetupClient creates a TSO client for test.
```

File: tests/mcs/testutil.go [hunk 2]
```diff
@@ -37,6 +37,13 @@ func SetupTSOClient(ctx context.Context, re *require.Assertions, endpoints []str
...
+func SetupClient(ctx context.Context, re *require.Assertions, endpoints []string, opts ...pd.ClientOption) pd.Client {
```
Comment (binshi-bing) [inline tests/mcs/testutil.go:41]: Shall we change the previous one SetupTSOClient to SetupClientWithKeyspace?

Comment (binshi-bing) [reply tests/mcs/testutil.go:41]: There is no specific TSO client now. Just the difference of w/ or w/o keyspace. I forgot to change SetupTSOClient to  SetupClientWithKeyspace.

**Review by binshi-bing — COMMENTED** [review]
left a few comments

File: tests/mcs/tso/tso_service_test.go [hunk 3]
```diff
@@ -39,8 +43,8 @@ type tsoServiceTestSuite struct {
 	cluster          *tests.TestCluster
 	pdLeader         *tests.TestServer
 	backendEndpoints string
-	tsoSvr1          *tsosvr.Server
-	tsoSvrCleanup1   func()
+	tsoSvr           *tsosvr.Server
```
[the remark this answers is not in the document]
Comment (rleungx) [reply tests/mcs/tso/tso_service_test.go:46]: When we need it, we can do the change.

File: tests/mcs/testutil.go [hunk 4]
```diff
@@ -37,6 +37,13 @@ func SetupTSOClient(ctx context.Context, re *require.Assertions, endpoints []str
...
+	cli, err := pd.NewClientWithContext(ctx, endpoints, pd.SecurityOption{}, opts...)
```
[the remark this answers is not in the document]
Comment (rleungx) [reply tests/mcs/testutil.go:42]: Shall we also retry the different paths according to the different errors?

**Review by binshi-bing — APPROVED** [review]
LGTM. Left a few comments in which some of them aren't necessary to address in this pr.

File: tests/tso/client_test.go [hunk 5]
```diff
@@ -167,3 +170,199 @@ func (suite *tsoClientTestSuite) TestUpdateAfterResetTSO() {
...
+					re.Less(lastTS, ts)
+					lastTS = ts
+				}
+			}
+		}()
+	}
+	wg.Wait()
+}
+
+func TestRandomShutdown(t *testing.T) {
```
Comment (JmPotato) [inline tests/tso/client_test.go:303]: Could this test be run for both two modes using the suite?

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**